reporting/PostgresBackend.cpp

```diff
@@ -378,70 +378,155 @@ PostgresBackend::fetchBookOffers(
     std::uint32_t limit,
     std::optional<ripple::uint256> const& cursor) const
 {
-    auto index = getBookIndexOfSeq(ledgerSequence);
-    if (!index)
-        return {};
-    PgQuery pgQuery(pgPool_);
-    ripple::uint256 zero = {};
-    std::optional<std::string> warning;
+    auto rng = fetchLedgerRange();
+
+    if(!rng)
+        return {{},{}};
+
+    ripple::uint256 bookBase = 
+        ripple::keylet::quality({ripple::ltDIR_NODE, book}, 0).key;
+    ripple::uint256 bookEnd = ripple::getQualityNext(bookBase);
+
+    using bookKeyPair = std::pair<ripple::uint256, ripple::uint256>;
+    auto getBooks = [this, &bookBase, &bookEnd, &limit](std::uint32_t sequence)
+        -> std::pair<bool, std::vector<bookKeyPair>>
     {
+        BOOST_LOG_TRIVIAL(info) << __func__ << ": Fetching books between "
+                        << "0x" <<  ripple::strHex(bookBase) << " and " 
+                        << "0x" << ripple::strHex(bookEnd) << "at ledger "
+                        << std::to_string(sequence);
+
+        auto start = std::chrono::system_clock::now();
+
         std::stringstream sql;
-        sql << "SELECT offer_key FROM books WHERE book = "
-            << "\'\\x" << ripple::strHex(zero)
-            << "\' AND ledger_seq = " << std::to_string(*index);
+        sql << "SELECT COUNT(*) FROM books WHERE "
+            << "book = \'\\x" << ripple::strHex(ripple::uint256(beast::zero)) 
+            << "\' AND ledger_seq = " << std::to_string(sequence);
+
+        bool complete;
+        PgQuery pgQuery(this->pgPool_);
         auto res = pgQuery(sql.str().data());
-        sql << " ORDER BY offer_key ASC"
-            << " LIMIT " << std::to_string(limit);
         if (size_t numRows = checkResult(res, 1))
+            complete = res.asInt(0, 0) != 0;
+        else 
+            return {true, {}};
```

Comment:
I think we want to return false here, since we don't know if the flag ledger is complete.